client-src/elements/form-definition.js

```diff
@@ -85,378 +95,456 @@ export function formatFeatureForEdit(feature) {
-  [INTENT_STAGES.INTENT_SHIP[0]]: FLAT_PREPARE_TO_SHIP_FIELDS,
-  [INTENT_STAGES.INTENT_ROLLOUT[0]]: ENTERPRISE_PREPARE_TO_SHIP,
-  [INTENT_STAGES.INTENT_SHIPPED[0]]: FLAT_SHIP_FIELDS,
-  [INTENT_STAGES.INTENT_REMOVED[0]]: DEPRECATION_REMOVED,
+// Stage_type values for each process.  Even though some of the stages
+// in these processes are similar to each other, they have distinct enum
+// values so that they can have different gates.
+
+// For incubating new standard features: the "blink" process.
+const STAGE_BLINK_INCUBATE = 110;
+const STAGE_BLINK_PROTOTYPE = 120;
+const STAGE_BLINK_DEV_TRIAL = 130;
+const STAGE_BLINK_EVAL_READINESS = 140;
+const STAGE_BLINK_ORIGIN_TRIAL = 150;
+// const STAGE_BLINK_EXTEND_ORIGIN_TRIAL = 151;
+const STAGE_BLINK_SHIPPING = 160;
+// Note: We might define post-ship support stage(s) later.
+
+// For implementing existing standards: the "fast track" process.
+const STAGE_FAST_PROTOTYPE = 220;
+const STAGE_FAST_DEV_TRIAL = 230;
+const STAGE_FAST_ORIGIN_TRIAL = 250;
+// const STAGE_FAST_EXTEND_ORIGIN_TRIAL = 251;
+const STAGE_FAST_SHIPPING = 260;
+
+// For developer-facing code changes not impacting a standard: the "PSA" process.
+const STAGE_PSA_IMPLEMENT_FIELDS = 320;
+const STAGE_PSA_DEV_TRIAL = 330;
+const STAGE_PSA_SHIPPING = 360;
+
+// For deprecating a feature: the "DEP" process.
+// const STAGE_DEP_PLAN = 410;
+const STAGE_DEP_DEV_TRIAL = 430;
+const STAGE_DEP_DEPRECATION_TRIAL = 450;
+// const STAGE_DEP_EXTEND_DEPRECATION_TRIAL = 451;
+const STAGE_DEP_SHIPPING = 460;
+// const STAGE_DEP_REMOVE_CODE = 470;
+
+// Note STAGE_* enum values 500-999 are reseverd for future WP processes.
+
+// Define enterprise feature processes.
+// Note: This stage can ge added to any feature that is following any process.
+const STAGE_ENT_ROLLOUT = 1061;
+const STAGE_ENT_SHIPPED = 1070;
+
+// TODO(danielrsmith): Differentiate origin trial and trial extension fields.
+export const FORMS_BY_STAGE_TYPE = {
+  [STAGE_BLINK_INCUBATE]: FLAT_INCUBATE_FIELDS,
+  [STAGE_BLINK_PROTOTYPE]: FLAT_PROTOTYPE_FIELDS,
+  [STAGE_BLINK_DEV_TRIAL]: FLAT_DEV_TRIAL_FIELDS,
+  [STAGE_BLINK_EVAL_READINESS]: FLAT_EVAL_READINESS_TO_SHIP_FIELDS,
+  [STAGE_BLINK_ORIGIN_TRIAL]: FLAT_ORIGIN_TRIAL_FIELDS,
+  [STAGE_BLINK_SHIPPING]: FLAT_PREPARE_TO_SHIP_FIELDS,
+
+  [STAGE_FAST_PROTOTYPE]: FLAT_PROTOTYPE_FIELDS,
+  [STAGE_FAST_DEV_TRIAL]: FLAT_DEV_TRIAL_FIELDS,
+  [STAGE_FAST_ORIGIN_TRIAL]: FLAT_ORIGIN_TRIAL_FIELDS,
+  [STAGE_FAST_SHIPPING]: FLAT_PREPARE_TO_SHIP_FIELDS,
+
+  [STAGE_PSA_IMPLEMENT_FIELDS]: PSA_IMPLEMENT_FIELDS,
+  [STAGE_PSA_DEV_TRIAL]: FLAT_DEV_TRIAL_FIELDS,
+  [STAGE_PSA_SHIPPING]: PSA_PREPARE_TO_SHIP_FIELDS,
+
+  [STAGE_DEP_DEV_TRIAL]: FLAT_DEV_TRIAL_FIELDS,
+  [STAGE_DEP_DEPRECATION_TRIAL]: DEPRECATION_ORIGIN_TRIAL_FIELDS,
+  [STAGE_DEP_SHIPPING]: DEPRECATION_PREPARE_TO_SHIP_FIELDS,
+  [STAGE_ENT_ROLLOUT]: FLAT_ENTERPRISE_PREPARE_TO_SHIP_FIELDS,
+  [STAGE_ENT_SHIPPED]: FLAT_PREPARE_TO_SHIP_FIELDS,
```

Comment:
This is not an ideal solution to mirror the enums server side, but maybe we can go with this for now and plan a better approach later on.